chains/evm/calls/transactor/monitored/monitored.go

```diff
@@ -0,0 +1,199 @@
+	gasPriceClient calls.GasPricer,
+	client calls.ClientDispatcher,
+	maxGasPrice *big.Int,
+	increaseFactor *big.Int,
+) *MonitoredTransactor {
+	return &MonitoredTransactor{
+		client:         client,
+		gasPriceClient: gasPriceClient,
+		txFabric:       txFabric,
+		pendingTxns:    make(map[common.Hash]RawTx),
+		maxGasPrice:    maxGasPrice,
+		increaseFactor: increaseFactor,
+	}
+}
+
+func (t *MonitoredTransactor) Transact(to *common.Address, data []byte, opts transactor.TransactOptions) (*common.Hash, error) {
+	t.client.LockNonce()
+	defer t.client.UnlockNonce()
+
+	n, err := t.client.UnsafeNonce()
+	if err != nil {
+		return &common.Hash{}, err
+	}
+
+	err = transactor.MergeTransactionOptions(&opts, &transactor.DefaultTransactionOptions)
+	if err != nil {
+		return &common.Hash{}, err
+	}
+
+	gp := []*big.Int{opts.GasPrice}
+	if opts.GasPrice.Cmp(big.NewInt(0)) == 0 {
+		gp, err = t.gasPriceClient.GasPrice(&opts.Priority)
+		if err != nil {
+			return &common.Hash{}, err
+		}
+	}
+
+	rawTx := RawTx{
+		to:           to,
+		nonce:        n.Uint64(),
+		value:        opts.Value,
+		gasLimit:     opts.GasLimit,
+		gasPrice:     gp,
+		data:         data,
+		submitTime:   time.Now(),
+		creationTime: time.Now(),
+	}
+	tx, err := t.txFabric(rawTx.nonce, rawTx.to, rawTx.value, rawTx.gasLimit, rawTx.gasPrice, rawTx.data)
+	if err != nil {
+		return &common.Hash{}, err
+	}
+
+	h, err := t.client.SignAndSendTransaction(context.TODO(), tx)
+	if err != nil {
+		return &common.Hash{}, err
+	}
+
+	t.txLock.Lock()
+	t.pendingTxns[h] = rawTx
+	t.txLock.Unlock()
+
+	err = t.client.UnsafeIncreaseNonce()
+	if err != nil {
+		return &common.Hash{}, err
+	}
+
+	return &h, nil
+}
+
+func (t *MonitoredTransactor) Monitor(
+	ctx context.Context,
+	resendInterval time.Duration,
+	txTimeout time.Duration,
+	tooNewTransaction time.Duration,
+) {
+	ticker := time.NewTicker(resendInterval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			{
+				t.txLock.Lock()
+				pendingTxCopy := make(map[common.Hash]RawTx, len(t.pendingTxns))
+				for k, v := range t.pendingTxns {
+					pendingTxCopy[k] = v
+				}
+				t.txLock.Unlock()
+
+				for oldHash, tx := range pendingTxCopy {
+					receipt, err := t.client.TransactionReceipt(context.Background(), oldHash)
+					if err == nil {
+						if receipt.Status == types.ReceiptStatusSuccessful {
+							log.Info().Msgf("Executed transaction %s with nonce %d", oldHash, tx.nonce)
+						} else {
+							log.Error().Msgf("Transaction %s failed on chain with nonce %d", oldHash, tx.nonce)
+						}
+
+						delete(t.pendingTxns, oldHash)
+						continue
+					}
+
+					if time.Since(tx.creationTime) > txTimeout {
+						log.Error().Msgf("Transaction %s with nonce %d has timed out", oldHash, tx.nonce)
+						delete(t.pendingTxns, oldHash)
+						continue
+					}
+					if time.Since(tx.submitTime) < tooNewTransaction {
+						continue
+					}
+
+					hash, err := t.resendTransaction(&tx)
```

Comment:
@mpetrun5 I think we should log (debug) when resending transaction, probably even better in function `resendTransaction` so we can log new gas params.